FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_demo_helpers.c

```diff
@@ -350,7 +355,11 @@ static TlsTransportStatus_t prvConnectToServerWithBackoffRetries( NetworkContext
     /* ALPN protocols must be a NULL-terminated list of strings. Therefore,
      * the first entry will contain the actual ALPN protocol string while the
      * second entry must remain NULL. */
-    char * pcAlpnProtocols[] = { NULL, NULL };
+    #ifdef democonfigCLIENT_USERNAME
```

Comment:
This decision should be performed near line 187 where the ALPN choices are defined.

Reply:
I disagree. Customers will likely want to adjust the selected alpn based on the port number or auth mechanism at runtime.

Reply:
I agree with that idea.  We should make the code work that way.